src/server/pfs/server/url.go

```diff
@@ -84,14 +88,45 @@ func putFileURL(ctx context.Context, taskService task.Service, uw *fileset.Unord
 
 func putFileURLRecursive(ctx context.Context, taskService task.Service, uw *fileset.UnorderedWriter, dst, tag string, src *pfs.AddFile_URLSource) error {
 	inputChan := make(chan *types.Any)
+	outputChan := make(chan *types.Any)
 	eg, ctx := errgroup.WithContext(ctx)
+	doer := taskService.NewDoer(URLTaskNamespace, uuid.NewWithoutDashes(), nil)
+	// Create tasks.
 	eg.Go(func() error {
-		// TODO: Add cache?
-		doer := taskService.NewDoer(URLTaskNamespace, uuid.NewWithoutDashes(), nil)
-		err := doer.Do(ctx, inputChan, func(_ int64, output *types.Any, err error) error {
+		if err := coordinateTasks(ctx, src.URL, func(startOffset, endOffset int64, startPath, endPath string) error {
+			input, err := serializePutFileURLTask(&PutFileURLTask{
+				Dst:         dst,
+				Datum:       tag,
+				URL:         src.URL,
+				StartOffset: startOffset,
+				EndOffset:   endOffset,
+				StartPath:   startPath,
+				EndPath:     endPath,
+			})
 			if err != nil {
 				return err
 			}
+			select {
+			case inputChan <- input:
+			case <-ctx.Done():
+				return errors.EnsureStack(ctx.Err())
+			}
+			return nil
+		}); err != nil {
+			return err
+		}
+		close(inputChan)
+		return nil
+	})
+	// Order output from input channel.
+	eg.Go(func() error {
+		// TODO: Add cache?
+		defer close(outputChan)
+		return task.DoOrdered(ctx, doer, inputChan, outputChan, 100)
```

Comment:
`DoOrdered` could just return the output through a callback. That is the underlying model of the task chain, which would make the outputChan and additional goroutine unnecessary.